R/makephyloseqFromTreeSummarizedExperiment.R

```diff
@@ -0,0 +1,113 @@
+#' Create a phyloseq object from a TreeSummarizedExperiment object
+#'
+#' @param x a \code{TreeSummarizedExperiment} object
+#'
+#' @param abund_values
+#' A single character value for selecting the
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{assay}}
+#'   to be included in the phyloseq object that is created.
+#'
+#'  @details
+#' \code{makePhyloseqFromTreeSummarizedExperiment} is used for creating a phyloseq
+#' object from TreeSummarizedExperiment object.
+#'
+#' @return
+#' An object of class \code{Phyloseq} object.
+#'
+#' @name makePhyloseqFromTreeSummarizedExperiment
+#' @export
+#'
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#'
+#' @examples
+#' # Get TSE object
+#' data("GlobalPatterns")
+#' TSE <- GlobalPatterns
+#'
+#' # Create a phyloseq object from it
+#' phy <- makePhyloseqFromTreeSummarizedExperiment(TSE)
+#'
+#' # By default the chosen table is counts, but if there is other tables,
+#' # they can be chosen with abund_values.
+#'
+#' # Counts relative abundances table
+#' TSE <- transformCounts(TSE, method = "relabundance")
+#' phy2 <- makePhyloseqFromTreeSummarizedExperiment(TSE, abund_values = "relabundance")
+#'
+NULL
+
+#' @rdname makePhyloseqFromTreeSummarizedExperiment
+#' @export
+setGeneric("makePhyloseqFromTreeSummarizedExperiment", signature = c("x"),
+           function(x, abund_values = "counts")
+               standardGeneric("makePhyloseqFromTreeSummarizedExperiment"))
+
+
+#' @rdname makePhyloseqFromTreeSummarizedExperiment
+#' @export
+setMethod("makePhyloseqFromTreeSummarizedExperiment", signature = c(x = "SummarizedExperiment"),
+          function(x, abund_values = "counts"){
+
+            # Input check
+            # Check object
+            if(!is(x,"TreeSummarizedExperiment")){
+                stop("'x' must be a 'TreeSummarizedExperiment' object")
+            }
+
+            # Check abund_values
+            .check_abund_values(abund_values, x)
+
+            # Get the assay from x
+            otu_table <- assay(x, abund_values)
+            # Convert to otu_table
+            otu_table <- phyloseq::otu_table(otu_table, taxa_are_rows = TRUE)
+
+            # If rowData has information, stores it to tax_table and converts is to phyloseq's tax_table.
+            # Otherwise, tax_table is NULL, and it is converted to phyloseq's tax_table.
+            if(nrow(rowData(x))>0 && ncol(rowData(x))>0){
+                tax_table <- as.matrix(rowData(x))
+                tax_table <- phyloseq::tax_table(tax_table)
+            } else {
+                tax_table <- NULL
+                tax_table <- phyloseq::tax_table(tax_table, errorIfNULL = FALSE)
+            }
```

Comment:
Does phyloseq support non-taxonomic data as tax_table? I don't think so, since it is a matrix and therefore can only be of type `character`.

It would be best to subset to taxonomic data only using `rowData(x)[,taxonomyRanks(x)]`

the else is also not necessary.

Reply:
Changed